sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -332,10 +360,12 @@ public long getSplitBacklogBytes() {
   private final String name;
   private @Nullable Consumer<byte[], byte[]> consumer = null;
   private final List<PartitionState<K, V>> partitionStates;
-  private @Nullable KafkaRecord<K, V> curRecord = null;
+  private @MonotonicNonNull KafkaRecord<K, V> curRecord = null;
```

Comment:
seems like curTimestamp would need this too

Reply:
It is not enforced by static analysis, but I think it is true that the value is monotonic non-null. Done.